### PR TITLE
Pagination for IRS Endpoint

### LINKED ISSUE #864: Add pagination to IRS
Using the same idea/format from #821, add pagination to the IRS endpoint.

### PR DESCRIPTION
**Pull request by schbetsy**
Closes #864 

Known issue: This PR doesn't update the `SubmissionIrsPathsSpec`. It is still commented out to avoid the `slick.util.AsyncExecutor` issue from previous sprints.

### CODE REVIEW

File: census/src/main/scala/hmda/census/model/CbsaLookup.scala [hunk 1]
```diff
@@ -34,8 +34,10 @@ object CbsaLookup extends CbsaResourceUtils {
     }
   }.toSeq
 
-  val codeMap: Map[String, String] = {
-    values.map(x => (x.cbsa, x.cbsaTitle)).toMap
+  def nameFor(cbsaId: String): String = {
+    values.find(_.cbsa == cbsaId)
```
Comment (nickgrippin) [inline census/src/main/scala/hmda/census/model/CbsaLookup.scala]: This new approach is interesting, is there a performance improvement gained by switching to `find`?

File: query/src/main/scala/hmda/query/model/filing/Msa.scala [hunk 2]
```diff
@@ -33,6 +16,40 @@ case class MsaWithName(
     homePurchase: Int,
     homeImprovement: Int,
     refinance: Int
+) {
+  def addName: MsaWithName = MsaWithName(
```
Comment (nickgrippin) [inline query/src/main/scala/hmda/query/model/filing/Msa.scala]: The return type on this method is pretty clear, I don't think it needs to be specified in the declaration.

File: census/src/main/scala/hmda/census/model/CbsaLookup.scala (hunk 1 above)
Comment (schbetsy) [inline census/src/main/scala/hmda/census/model/CbsaLookup.scala]: Just an improvement in modularity of code and separation of concerns. Don't know about performance.

File: query/src/main/scala/hmda/query/model/filing/Msa.scala (hunk 2 above)
Comment (schbetsy) [inline query/src/main/scala/hmda/query/model/filing/Msa.scala]: Is it important to remove it?

File: census/src/main/scala/hmda/census/model/CbsaLookup.scala (hunk 1 above)
Comment (schbetsy) [inline census/src/main/scala/hmda/census/model/CbsaLookup.scala]: I can add the map back in if we're concerned about performance.

File: query/src/main/scala/hmda/query/model/filing/Msa.scala (hunk 2 above)
Comment (jmarin) [inline query/src/main/scala/hmda/query/model/filing/Msa.scala]: In general, adding return types for methods is encouraged in typical `Scala` code.

File: api/src/main/scala/hmda/api/protocol/processing/MsaProtocol.scala [hunk 3]
```diff
@@ -1,11 +1,36 @@
...
+
+    override def read(json: JsValue): IrsResponse = {
+      json.asJsObject.getFields("msas", "summary", "total", "_links") match {
+        case Seq(JsArray(msas), JsObject(summ), JsNumber(tot), JsObject(links)) =>
+          val msaCollection: Seq[MsaWithName] = msas.map(_.convertTo[MsaWithName])
+          val path: String = PaginatedResponse.staticPath(links("href").convertTo[String])
+          val currentPage: Int = PaginatedResponse.currentPage(links("self").convertTo[String])
+          val summary: MsaSummary = JsObject(summ).convertTo[MsaSummary]
+          val total: Int = tot.intValue
+          IrsResponse(msaCollection.toList, summary, path, currentPage, total)
```
Comment (jmarin) [inline api/src/main/scala/hmda/api/protocol/processing/MsaProtocol.scala]: I would remove the types for these `val` declarations if the compiler is happy with it. In general, `val` or `var` declarations that can use type inference should do so.

File: census/src/main/scala/hmda/census/model/CbsaLookup.scala (hunk 1 above)
Comment (jmarin) [inline census/src/main/scala/hmda/census/model/CbsaLookup.scala]: I have a strong opinion about "performance". In the absence of an objective measurement that proves a performance issue, there isn't a performance issue. So I'm ok with this change based on what I see in the code

File: api/src/main/scala/hmda/api/protocol/processing/MsaProtocol.scala (hunk 3 above)
Comment (schbetsy) [inline api/src/main/scala/hmda/api/protocol/processing/MsaProtocol.scala]: Interesting. I'm fine with that (and the compiler seems to be too). Done in 8b4a628a46e6915f35ac7df23278501a30d0d983.

**Pull request merged**